mesonwrap/tempfile.py

```diff
@@ -0,0 +1,29 @@
+# Let's try to make it drop-in replacement for tempfile module
+import platform as _platform
+from tempfile import *
+
+from mesonwrap import tempfile_helper as _tempfile_helper
+
+
+# It is not desirable to maintain completely separate implementation
+# of TemporaryDirectory, however it is beneficial for Windows users
+# that have antivirus software that interferes with file deletion.
+# As a compromise the code below tries to override standard TemporaryDirectory
+# cleanup() method in order to hook _tempfile_helper.windows_proof_rmtree().
+# Since _finalizer is not public this implementation may partially break
+# in the future, but expected failure mode in this case is just a double
+# finalization, hence possible warning.
+class _WindowsTemporaryDirectory(TemporaryDirectory):
```

Comment:
__[Flake8]__

[F405] 'TemporaryDirectory' may be undefined, or defined from star imports: tempfile


<!-- Comment by [Sider](https://sider.review) -->